app/views/kudos/index.html.erb

```diff
@@ -20,3 +21,23 @@
     <% end %>
   <% end %>
 </div>
+<% end %>
+<% if @kudos %>
+<!--Descriptive page name, messages and instructions-->
+<h2 class="heading"><%= ts('Kudos History') %></h2>
+<!--/descriptions-->
+<!--main content-->
+<% if logged_in? && @kudos.present? %>
```

Comment:
OK but then that is a bug in the check you're doing in the controller and you need to fix it there. Just checking that you're logged in would not prevent this anyway--that would only prevent guests from seeing the kudos for a user.